core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -175,6 +176,62 @@
 
 public class InternalEngineTests extends EngineTestCase {
 
+    public void testVersionMapAfterAutoIDDocument() throws IOException {
+        ParsedDocument doc = testParsedDocument("1", null, testDocumentWithTextField(),
+            new BytesArray("{}".getBytes(Charset.defaultCharset())), null);
+        Engine.Index operation = appendOnlyReplica(doc, false, 1, randomIntBetween(0, 5));
```

Comment:
should we randomize between primary and replica? they slightly different code paths in the engine.